### PR TITLE
Fix/asub 8551 apple signin redirect

### PR DESCRIPTION
**Pull request by accbjt**
**Be sure to run `npm test`, `npm run lint`, and write detailed test steps before requesting reviewers**

#### Jira Ticket: [THEMES-](https://arcpublishing.atlassian.net/browse/THEMES-)
- https://arcpublishing.atlassian.net/browse/ASUB-8551

_Information about what you changed for this PR. Include any dependencies, companion PRs, relevant screenshots, or config changes._

## Test Steps

_Add detailed test steps a reviewer must complete to test this PR_

Video recording of testing this locally. I created this so you don't have to set this up on your local machine. This depends heavily on the Apple sign on implementation on subssinglesite and is decent amount of work to get it setup locally.

https://www.loom.com/share/2b1bbc5bd69348c9ac3e05b1c09a4f4a?sid=194460fd-1cee-46d5-8ea8-1c2dd085e698

Here is another example on an article page.

https://www.loom.com/share/c0a83222672a4ab885ff7a4d1b26a278?sid=0608528b-9acf-42c0-a87c-41f2b5b0cf71

### CODE REVIEW

File: blocks/identity-block/components/login/index.jsx [hunk 1]
```diff
@@ -57,13 +67,15 @@ const useLogin = ({
 			const redirectUrlLocation = new URL(document.referrer);
 
 			if (searchParams.has('reset_password')) {
-				setCurrentRedirectToURL(`${redirectURL}${redirectUrlLocation.search}`);
+				const newRedirectUrl = `${redirectURL}${redirectUrlLocation.search}`;
+
+				setRedirectUrl(newRedirectUrl);
```
Comment (LauraPinilla) [inline blocks/identity-block/components/login/index.jsx]: @accbjt  I think we can define a `let newRedirectUrl` and move this line out from the if-else statement.

Comment (accbjt) [reply blocks/identity-block/components/login/index.jsx]: @LauraPinilla Okay, I made those updates.

File: blocks/identity-block/utils/validate-redirect-url.js [hunk 2]
```diff
@@ -13,6 +13,7 @@ const validateURL = (url) => {
 		return url;
 	}
 
+	sessionStorage.setItem("ArcXP_redirectUrl", "/");
```
Comment (LauraPinilla) [inline blocks/identity-block/utils/validate-redirect-url.js:16]: Hi @accbjt, why are you setting ArcXP_redirectUrl to "/". Sounds like the value we are setting here should depend on the custom fields or the page in which the user is.

Comment (accbjt) [reply blocks/identity-block/utils/validate-redirect-url.js:16]: @LauraPinilla The original `validateURL` return '/' if the url is not valid so I was just following what they set in that function. I think if the url is not a valid url then it is okay to set it as `/`. That is the original logic and I don't want to change that and make it more complicated.

**Review by LauraPinilla — DISMISSED** [review]
lgtm!

File: blocks/identity-block/components/social-sign-on/utils/useSocialSignIn.js [hunk 3]
```diff
@@ -21,6 +21,7 @@ function useSocialSignIn(redirectURL, isOIDC, socialSignOnIn, onError = () => {}
 					loginByOIDC();
 				} else {
 					const validatedURL = validateURL(redirectURL);
+
```
Comment (malavikakoppula) [inline blocks/identity-block/components/social-sign-on/utils/useSocialSignIn.js]: @accbjt Can you please revert this new line

File: blocks/identity-block/components/social-sign-on/utils/useSocialSignIn.js [hunk 4]
```diff
@@ -51,6 +52,7 @@ function useSocialSignIn(redirectURL, isOIDC, socialSignOnIn, onError = () => {}
 							loginByOIDC();
 						} else {
 							const validatedURL = validateURL(redirectURL);
+
```
Comment (malavikakoppula) [inline blocks/identity-block/components/social-sign-on/utils/useSocialSignIn.js]: @accbjt Same here

File: blocks/identity-block/features/account-management/default.jsx [hunk 5]
```diff
@@ -43,6 +43,7 @@ function AccountManagement({ customFields }) {
 		const checkLoggedInStatus = () =>
 			Identity.isLoggedIn().then((isLoggedIn) => {
 				if (!isLoggedIn) {
+
```
Comment (malavikakoppula) [inline blocks/identity-block/features/account-management/default.jsx]: @accbjt same here

File: blocks/identity-block/features/login/default.jsx [hunk 6]
```diff
@@ -90,6 +90,7 @@ const Login = ({ customFields }) => {
 							loginByOIDC();
 						} else {
 							const validatedURL = validateURL(loginRedirect);
+
```
Comment (malavikakoppula) [inline blocks/identity-block/features/login/default.jsx]: @accbjt same here

**Review by malavikakoppula — CHANGES_REQUESTED** [review]
@accbjt very small update removing newlines in multiple components. Everything else is look good

**Comment by accbjt** [comment]
@malavikakoppula Okay, I made those changes. Thanks

**Review by malavikakoppula — APPROVED** [review]
LGTM

**Pull request merged**